app/screens/reschedule_draft/reschedule_draft.test.tsx

```diff
@@ -0,0 +1,197 @@
+import {renderWithEverything} from '@test/intl-test-helper';
+import TestHelper from '@test/test_helper';
+
+import RescheduledDraft from './reschedule_draft';
+
+import type {Database} from '@nozbe/watermelondb';
+import type ScheduledPostModel from '@typings/database/models/servers/scheduled_post';
+
+jest.mock('@actions/remote/scheduled_post', () => ({
+    updateScheduledPost: jest.fn().mockResolvedValue({scheduledPost: {} as ScheduledPost, error: undefined}),
+}));
+
+jest.mock('@screens/navigation', () => ({
+    buildNavigationButton: jest.fn().mockReturnValue({
+        id: 'reschedule-draft',
+        testID: 'reschedule-draft.save.button',
+        showAsAction: 'always',
+    }),
+    dismissModal: jest.fn(),
+    setButtons: jest.fn(),
+}));
+
+jest.mock('react-native-navigation', () => {
+    const registerComponentListenerMock = jest.fn();
+    return {
+        Navigation: {
+            events: () => ({
+                registerComponentListener: registerComponentListenerMock,
+            }),
+        },
+    };
+});
+
+jest.mock('@context/server', () => ({
+    useServerUrl: jest.fn(),
+}));
+
+const SERVER_URL = 'https://appv1.mattermost.com';
+
+describe('RescheduledDraft', () => {
+    let database: Database;
+
+    const mockDraft = {
+        scheduledAt: moment().add(1, 'day').valueOf(),
+        toApi: jest.fn().mockResolvedValue({
+            scheduled_at: moment().add(1, 'day').valueOf(),
+        }),
+    } as unknown as ScheduledPostModel;
+
+    const baseProps = {
+        componentId: Screens.RESCHEDULE_DRAFT,
+        closeButtonId: 'close-button-id',
+        currentUserTimezone: {
+            useAutomaticTimezone: true,
+            automaticTimezone: 'America/New_York',
+            manualTimezone: '',
+        },
+        draft: mockDraft,
+    };
+
+    beforeAll(async () => {
+        const server = await TestHelper.setupServerDatabase();
+        database = server.database;
+    });
+
+    beforeEach(() => {
+        jest.mocked(useServerUrl).mockReturnValue(SERVER_URL);
+    });
+
+    it('should render correctly', () => {
+        const {getByTestId} = renderWithEverything(
+            <RescheduledDraft {...baseProps}/>, {database},
+        );
+
+        expect(getByTestId('edit_post.screen')).toBeTruthy();
+    });
+
+    it('should have navigation component registered on initialization', () => {
+        renderWithEverything(<RescheduledDraft {...baseProps}/>, {database});
+
+        // Verify navigation listener was registered
+        expect(Navigation.events().registerComponentListener).toHaveBeenCalledWith(
+            expect.any(Object),
+            baseProps.componentId,
+        );
+
+        // Check that the component registered with proper navigation handler
+        const registerCall = jest.mocked(Navigation.events().registerComponentListener).mock.calls[0][0];
+        expect(registerCall).toBeDefined();
+        expect(registerCall.navigationButtonPressed).toBeDefined();
```

Comment:
We check it exist. Do we want to check if it does what it has to do? I would expect something similar to what you have in the last test of this file, but also for other buttons (not only the close one).

Reply:
Added what happens when it clicked on save button.